packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -943,31 +945,35 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     options?:
       | {
           TEMPORARY__keepSectionIndex?: boolean | undefined;
+          skipPostProcessingEditing?: boolean | undefined;
         }
       | undefined,
   ): GeneratorFn<void> {
-    yield Promise.all(
-      inputs.flatMap((input) =>
-        input.data.elements.map((el) =>
-          promisify(() =>
-            runInAction(() => {
-              const element = graph.getElement(el.path);
-              if (input.parentElementPath) {
-                element.generationParentElement = graph.getElement(
-                  input.parentElementPath,
+    if (!options?.skipPostProcessingEditing) {
+      yield Promise.all(
+        inputs.flatMap((input) =>
+          input.data.elements.map((el) =>
+            promisify(() =>
+              runInAction(() => {
+                // TODO: check performance once `getNullableElement` is optimized
+                const element = graph.getElement(el.path);
```

Comment:
Wouldn't this mess up `generation` since you no longer know the parent?

Reply:
It will mess up the generated element editor in the sense that users won't be able to visit the parent. Have fixed this by enabling this in viewer mode and main edit mode.